llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -1575,16 +1575,25 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
         return eraseInstFromFunction(CI);
     }
 
+    bool SrcIsNull = false;
     // If we can determine a pointer alignment that is bigger than currently
     // set, update the alignment.
     if (auto *MTI = dyn_cast<AnyMemTransferInst>(MI)) {
       if (Instruction *I = SimplifyAnyMemTransfer(MTI))
         return I;
+      SrcIsNull = isa<ConstantPointerNull>(MTI->getRawSource());
     } else if (auto *MSI = dyn_cast<AnyMemSetInst>(MI)) {
       if (Instruction *I = SimplifyAnyMemSet(MSI))
         return I;
     }
 
+    // If src/dest is null, this memory intrinsic must be a noop.
+    if (!NullPointerIsDefined(MI->getFunction()) &&
```

Comment:
Don't you also have to pass the address space?

Reply:
Oops, parameter `AS` has a default argument :(
```
bool NullPointerIsDefined(const Function *F, unsigned AS = 0);
```